### ISSUE TITLE
Do not warn useless_conversion for (x..y).into_iter() which become necessary in the future edition?

### ISSUE THREAD

**Issue by anatawa12** [rust-lang/rust-clippy#17295]
This is relates to false positive but it's discussing if clippy should not report warning or not so I wrote this without bug template, but with all information in bug template.

New range types are stable and It's planned to make`(x..y)` be new range type in the next edition.
Therefore, accessing iterator methods directly on `(x..y)` looks not good to me so I wrote `(x..y).into_iter()`.
However, clippy generates useless_conversion warning for `(x..y).into_iter()`.

I don't know if clippy prefers future compatible syntax vs current edition only but less code so I'm creating issue as normal issue.

Reproducer

```rust
let peekable = (0..num_params).into_iter().peekable();
```

I saw this happen:

```
useless conversion to the same type: `std::ops::Range<usize>`
Help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.96.0/index.html#useless_conversion
Note: `#[warn(clippy::useless_conversion)]` on by default
Help: consider removing `.into_iter()`
```

Rust version

```
$ rustc -Vv
rustc 1.96.0 (ac68faa20 2026-05-25)
binary: rustc
commit-hash: ac68faa20c58cbccd01ee7208bf3b6e93a7d7f96
commit-date: 2026-05-25
host: aarch64-apple-darwin
release: 1.96.0
LLVM version: 22.1.2
$ 
$ clippy-driver -Vv
rustc 1.96.0 (ac68faa20 2026-05-25)
binary: rustc
commit-hash: ac68faa20c58cbccd01ee7208bf3b6e93a7d7f96
commit-date: 2026-05-25
host: aarch64-apple-darwin
release: 1.96.0
LLVM version: 22.1.2
```

**Comment by Jarcho** [comment]
We definitely need to stop warning on this. Once the `new_range` feature is stable we could lint these calls on a recent enough edition.
